server/mediaserver.go

```diff
@@ -559,7 +563,8 @@ func (s *LivepeerServer) HandlePush(w http.ResponseWriter, r *http.Request) {
 			return
 		}
 		st := stream.NewBasicRTMPVideoStream(appData)
-		cxn, err = s.registerConnection(st)
+		resolution := r.Header.Get("Content-Resolution")
+		cxn, err = s.registerConnection(st, resolution)
```

Comment:
A less invasive way to do this might be to add a `resolution` field within the streamParameters struct and using that instead: Something like:

1. Add a `resolution` field to the `streamParameters` struct
2. Set the resolution unconditionally within `createRTMPStreamIDHandler` . This might end up being `0x0` for HLS push. That's OK.
3. Within the HLS push function, re-assign `appData.resolution = r.Header.Get("Content-Resolution")` after calling `createRTMPStreamIDHandler`
4. Within `registerConnection`, set `Resolution: appData.Resolution`

How does that sound?

The LPMS stream API is starting to feel a bit clunky (ideally we'd be able to slot in different stream types, and hence supply our own `Width` / `Height` accessors) but this should be OK for now.

Reply:
That does sound a lot cleaner! However, wouldn't we need something like `appData.SetResolution(r.Header.Get("Content-Resolution"))` to set the resolution on `streamParameters`? The `AppData` interface is defined in `lmps > stream > interface.go`, so we'd have to add a `SetResolution() string` method there to access the `streamParameters` property, which would require LPMS editing. Is that something we'd want to do? 
```golang
func (s *streamParameters) SetResolution(resolution string) {
	s.resolution = resolution
}
```

Reply:
We'd also need this: 
```golang
func (s *streamParameters) Resolution() string {
	return s.resolution
}
```

Reply:
Reify the appData into a concrete type via `streamParams` and use that:

https://github.com/livepeer/go-livepeer/blob/f5106de234ff99f26916f6965a0f8fe41e789f25/server/mediaserver.go#L245-L253

```golang
appData := (createRTMPStreamIDHandler(s))(r.URL)
st := stream.NewBasicRTMPVideoStream(appData)
params := streamParams(st)
params.resolution = r.Header.Get("Content-Resolution")
cxn, err := s.registerConnection(st)
...
```

Reply:
OH I see. That's a much cleaner way of doing it. Thanks @j0sh !